lib/helpers/error-handler.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const ApiSaveError = require('../api-save-error');
+
+const JSONSanitizer = require('./json-sanitizer');
+
+class ErrorHandler {
+
+	/**
+	 * Handle an error for a model operation
+	 * Evaluates the error then set the respective response code and message
+	 * @param {API} apiInstance API instance (for setting response codes)
+	 * @param {Error} err The error to handle
+	 * @throws The received error parsed
+	 */
+	static handle(apiInstance, err) {
+
+		if(!this._isDuplicatedKey(err))
+			throw new ApiSaveError(err, ApiSaveError.codes.INTERNAL_ERROR);
+
+		this._handleDuplicatedKeyError(apiInstance, err);
+	}
+
+	static _isDuplicatedKey(err) {
+		return /E11000 duplicate key error/g.test(err.message);
+	}
+
+	static _handleDuplicatedKeyError(apiInstance, err) {
+
+		apiInstance.setCode(400);
+
+		let [fields] = err.message.match(/{.*}/g);
+
+		fields = JSONSanitizer.sanitizeAndParse(fields);
+
+		fields = Object.keys(fields)
+			.toString()
+			.replace(/,/g, ', ');
+
+		err.message = `A document for field or fields: '${fields}' already exists`;
```

Comment:
Podemos lograr que el mensaje final sea: _A document for field or fields: 'client', 'name' already exists_?